cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/runtimejob/DataprocRuntimeJobManager.java

```diff
@@ -384,7 +385,20 @@ private boolean isDeleteLifecycleEnabled(String bucketName) {
       }
       if (rule.getAction() instanceof BucketInfo.LifecycleRule.DeleteLifecycleAction &&
         rule.getCondition().getDaysSinceCustomTime() > 0) {
-        return true;
+        try {
+          int artifactsComputeHashTimeBucketDays = Integer.parseInt(
+            provisionerContext.getProperties().get(DataprocUtils.ARTIFACTS_COMPUTE_HASH_TIME_BUCKET_DAYS));
+          boolean isValid = rule.getCondition().getDaysSinceCustomTime() > artifactsComputeHashTimeBucketDays;
+          if (!isValid) {
+            LOG.warn("Bucket Lifecycle Rule for DeleteLifecycleAction is having lesser value than " +
```

Comment:
Days since custom time rule of delete lifecycle for bucket {} should be strictly greater than {} days